### DISCUSSION TITLE
UI design idea

### DISCUSSION THREAD

**Discussion by vladstudio** [mikekazakov/nimble-commander#132] [Ideas]
So I spent a bit of time trying out Nimble Commander, and there are some features that I reaaaally like (but this post is not about them :-) ).

As a UX/UI designer, I know no one asked, but I'd love to propose a brave redesign for NC – just to show some ideas of what's possible. 

<img width="1049" alt="image" src="https://github.com/mikekazakov/nimble-commander/assets/914320/6f561778-fd41-4e15-b0da-d6ba24d17338">

I laid out all the borders, margin and spacings according to what I think would work best.

The entire UI would be built from 8 colours, defined by a theme:

<img width="1284" alt="image" src="https://github.com/mikekazakov/nimble-commander/assets/914320/92516a6a-76c8-4445-b9a1-3e6eccb33948">

For example, here are another set of these 8 colors, literally takes 10 minutes to create a new theme:

<img width="1264" alt="image" src="https://github.com/mikekazakov/nimble-commander/assets/914320/fc0997b8-1c08-49cc-bed3-5fd3dc5e454e">

I also suggest that fonts are taken out of theme JSON, a theme only defines colours, and fonts are defined in Settings. Reason: the choice of fonts, unlike colours, should depend on what user has installed on their device.

Here's the Figma file if you're curious:

https://www.figma.com/file/Bm4vhN0Fr1448Z4EVVs74y/Nimble-Commander?type=design&node-id=17%3A2&mode=design&t=0OJX8NLBU0LdPz00-1

**Comment by vladstudio** [comment]
could not resist, another theme:

<img width="1258" alt="image" src="https://github.com/mikekazakov/nimble-commander/assets/914320/b736a947-f12f-483c-917c-629f98e3227f">

based on https://stephango.com/flexoki

**Comment by WWWFWWW** [comment]
Do you mean simplifying NC's Settings/Themes dialogue to be based around those eight categories only?

While I might agree that the current Themes editing inside NC can be a bit confusing (and creating a wholly new user theme might be better done in some GUI json editor with UI/colour preview, if such a thing exists), any simplified UI for editing themes like that based on a limited amount of categories would still have to take into account the advanced stuff themes offer, for example "syntax" filename highlighting rules, which is a pretty powerful tool.

I loved file highlighting based on rules in the old orthodox file commanders of yore, and NC offers the same if not better (without delving into ini files).

IIRC NC currently supports colouring filenames by extension mask and _attributes_, so I can have .exe, .com some colour (e.g. for browsing Wine prefixes or mounted Win shares), Mac executables a different colour, folders a brighter colour (though allowing fonts like bold would be nicer), hidden files a darker colour (or even 50% opacity), different graphic files a different colour, et cetera.

So any simplifying to something like eight main categories would still have to deal with all those other colours seamlessly.

**Reply by vladstudio** [reply]
Agreed – I only scratched the surface of theming, and did touch advanced topics, like syntax highlighting or colouring filenames by properties.

**Reply by mikekazakov** [reply]
Ohh, nice to hear to somebody uses this functionality! :)
Never thought about tying to it font properties like bold/italic, this can be quite powerful indeed, though such font galore could also lead to visual mess.

**Comment by mikekazakov** [comment]
I really like this clean design, thank you!
Nowadays NC's appearance feels somewhat outdated - it was essentially laid out around v1.0.7, i.e. ~9 years ago and it shows.
I'm open to refreshing the UI's feel and look, but realistically it depends on the scope of changes and whether there would be any volunteers to get their hands dirty in the source code :)
